WordPress/src/main/java/org/wordpress/android/ui/main/ChooseSiteActivity.kt

```diff
@@ -0,0 +1,398 @@
+                            ActivityUtils.hideKeyboardForced(searchView)
+                        }
+                        return true
+                    }
+
+                    override fun onQueryTextChange(newText: String): Boolean {
+                        AnalyticsTracker.track(Stat.SITE_SWITCHER_SEARCH_PERFORMED)
+                        viewModel.searchSites(newText, mode)
+                        return true
+                    }
+                })
+                return true
+            }
+
+            override fun onMenuItemActionCollapse(item: MenuItem): Boolean {
+                searchView.setOnQueryTextListener(null)
+                viewModel.loadSites(mode)
+                menuEditPin.isVisible = true
+                invalidateOptionsMenu()
+                return true
+            }
+        })
+    }
+
+    @Suppress("ReturnCount")
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.menu_pin -> {
+                if (adapter.mode is ActionMode.Pin) {
+                    disablePinSitesMode()
+                } else {
+                    enablePinSitesMode()
+                }
+                return true
+            }
+
+            R.id.menu_search -> return true
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+    /**
+     * Enable pin sites mode via menu items
+     */
+    private fun enablePinSitesMode() {
+        menuSearch.isVisible = false
+        menuEditPin.setIcon(null)
+        menuEditPin.title = "Done"
+        adapter.setActionMode(ActionMode.Pin)
+    }
+
+    /**
+     * Disable pin sites mode via menu items
+     */
+    private fun disablePinSitesMode() {
+        menuSearch.isVisible = true
+        menuEditPin.setIcon(R.drawable.pin_filled)
+        menuEditPin.title = "Edit Pins"
+        adapter.setActionMode(ActionMode.None)
+    }
+
+    private fun setupRecycleView() {
+        binding.recyclerView.layoutManager = LinearLayoutManager(this)
+        binding.recyclerView.adapter = adapter.apply {
+            onReload = { viewModel.loadSites(this@ChooseSiteActivity.mode) }
+            onSiteClicked = { selectSite(it) }
+            onSiteLongClicked = { onSiteLongClick(it) }
+        }
+        binding.recyclerView.scrollBarStyle = View.SCROLLBARS_OUTSIDE_OVERLAY
+        binding.recyclerView.setEmptyView(binding.actionableEmptyView)
+    }
+
+    private fun selectSite(siteRecord: SiteRecord) {
+        AppPrefs.addRecentlyPickedSiteId(siteRecord.localId)
+        setResult(RESULT_OK, Intent().putExtra(KEY_SITE_LOCAL_ID, siteRecord.localId))
+        finish()
+    }
+
+    private fun onSiteLongClick(siteRecord: SiteRecord) {
+        val site: SiteModel = siteStore.getSiteByLocalId(siteRecord.localId) ?: return
+        if (site.isUsingWpComRestApi.not()) {
+            showRemoveSelfHostedSiteDialog(site)
```

Comment:
I copied the logic from SitePickerActivity, but this is not triggered when I long clicked on my self-hosted site.

Reply:
This worked as expected for me both in the old and new implementation. I've tested with a self-hosted JN site
![Screenshot_20240326_152145](https://github.com/wordpress-mobile/WordPress-Android/assets/304044/ab65dfc6-4652-4a6f-a4ba-d552bee5ebc1)

Reply:
I'm not able to reproduce this with the latest version of the PR 🎉